tests/src/test/java/org/eclipse/hono/tests/registry/DeviceManagementIT.java

```diff
@@ -423,6 +430,209 @@ public void testDeregisterDeviceFailsForNonExistingDevice(final VertxTestContext
             .onComplete(ctx.completing());
     }
 
+    /**
+     * Tests verifying the search devices operation.
+     *
+     * @see <a href="https://www.eclipse.org/hono/docs/api/management/#/devices/searchDevicesForTenant"> 
+     *      Device Registry Management API - Search Devices</a>
+     */
+    @Nested
+    @EnabledIfSystemProperty(named = "hono.deviceregistry.supportsSearchDevices", matches = "true")
+    class SearchDevicesIT {
+        /**
+         * Verifies that a request to search devices fails with a {@value HttpURLConnection#HTTP_NOT_FOUND}
+         * when no matching devices are found.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesFailsWhenNoDevicesAreFound(final VertxTestContext ctx) {
+            final Device device = new Device().setEnabled(false);
+            final String filterJson = getFilterJson("/enabled", true, "eq");
+
+            registry.registerDevice(tenantId, deviceId, device)
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.empty(), Optional.empty(),
+                            List.of(filterJson), null, HttpURLConnection.HTTP_NOT_FOUND))
+                    .onComplete(ctx.completing());
+        }
+
+        /**
+         * Verifies that a request to search devices fails when page size is invalid.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesFailsForInvalidPageSize(final VertxTestContext ctx) {
+            final int invalidPageSize = -100;
+
+            registry.registerDevice(tenantId, deviceId)
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.of(invalidPageSize), Optional.empty(),
+                            null, null, HttpURLConnection.HTTP_BAD_REQUEST))
+                    .onComplete(ctx.completing());
+        }
+
+        /**
+         * Verifies that a request to search devices with a valid page size.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesWithPageSize(final VertxTestContext ctx) {
+            final int pageSize = 1;
+
+            CompositeFuture.all(registry.registerDevice(tenantId, new Device()), registry
+                    .registerDevice(tenantId, new Device())
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.of(pageSize), Optional.empty(), null, null,
+                            HttpURLConnection.HTTP_OK))
+                    .onComplete(ctx.succeeding(httpResponse -> {
+                        ctx.verify(() -> {
+                            final JsonArray response = httpResponse.bodyAsJsonArray();
+                            assertThat(response.size()).isEqualTo(1);
```

Comment:
how about also asserting the overall result set size?

Reply:
Here the result set size is being asserted to the page size. I didn't get it what do you mean by overall result set size?

Reply:
well, you have registered two devices and the first page of the search result contains the first match (page size = 1). However, the overall result set size should be 2 because you haven't specified any search criteria, right? So we could also check the *total* property in the response body. But wait a second, it looks like the response body currently doesn't contain the *total* and *result* properties but instead only contains the result set as a JSON array. IMHO this is currently not compliant with the spec ...

Reply:
I see what do you mean now. My bad I failed to notice the change in the result structure of the search devices operation and I followed the old one. I will create a separate PR to comply with the spec and then continue with the integration tests here.

Reply:
sounds good. No worries, I missed it too while reviewing the MongoDB registry implementation ...

Reply:
I have created  a PR #2150 to make the search devices result in compliant wit the spec.

Reply:
The overall size value is now being asserted and changes are made in accordance with the PR #2150 (Fix search devices result). I have pushed a separate commit for that.